x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/UnresolvedAttribute.java

```diff
@@ -119,16 +119,17 @@ public static String errorMessage(String name, List<String> potentialMatches) {
     }
 
     @Override
+    @SuppressWarnings("checkstyle:EqualsHashCode")// equals is implemented in parent. See innerEquals instead
     public int hashCode() {
         return Objects.hash(super.hashCode(), resolutionMetadata, unresolvedMsg);
     }
 
     @Override
-    public boolean equals(Object obj) {
-        if (super.equals(obj)) {
-            UnresolvedAttribute ua = (UnresolvedAttribute) obj;
-            return Objects.equals(resolutionMetadata, ua.resolutionMetadata) && Objects.equals(unresolvedMsg, ua.unresolvedMsg);
-        }
-        return false;
+    protected boolean innerEquals(Object o) {
+        var other = (UnresolvedAttribute) o;
+        return super.innerEquals(other)
+            && customMessage == other.customMessage
+            && Objects.equals(unresolvedMsg, other.unresolvedMsg)
+            && Objects.equals(resolutionMetadata, other.resolutionMetadata);
```

Comment:
The order between the checks for `resolutionMetadata` and `unresolvedMsg` were flipped - is that on purpose? (Not sure which one could supposedly be faster.)

Reply:
Fixed